src/algebra/lattice_ordered_group.lean

```diff
@@ -125,26 +125,29 @@ calc a⊓b * (a ⊔ b) = a ⊓ b * ((a * b) * (b⁻¹ ⊔ a⁻¹)) :
 
 namespace lattice_ordered_comm_group
 
--- Bourbaki A.VI.12 Definition 4
 /--
+Let `α` be a lattice ordered commutative group with identity `1`. For an element `a` of type `α`,
+the element `a ⊔ 1` is said to be the *positive component* of `a`, denoted `a⁺`.
+-/
+@[to_additive /-"
 Let `α` be a lattice ordered commutative group with identity `0`. For an element `a` of type `α`,
 the element `a ⊔ 0` is said to be the *positive component* of `a`, denoted `a⁺`.
--/
-@[to_additive pos
-  "Let `α` be a lattice ordered commutative group with identity `0`. For an element `a` of type `α`,
-  the element `a ⊔ 0` is said to be the *positive component* of `a`, denoted `a⁺`."]
-def mpos (a : α) : α :=  a ⊔ 1
-postfix `⁺`:1000 := mpos
+"-/,
+priority 100
+] -- see Note [lower instance priority]
+instance has_one_lattice_has_pos_part : has_pos_part (α)  := ⟨λa, a ⊔ 1⟩
```

Comment:
Can you add a lemma stating that `a⁺ = a ⊔ 1` and rewrite with those instead of `unfold has_pos_part.pos` (that will be more robust if the implementation changes)